### PR TITLE
TestCafe Integration Tests

### PR DESCRIPTION
**Pull request by slowr**
### Description of PR

Introduce TestCafe integration tests on CI.

What component(s) does this PR affect?

- [ ] Back-End (Database, Microservices, Containers, etc)
- [x] Front-End (Flask, API, etc)
- [ ] Docs
- [x] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

### Related Issue

Resolves #

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by slowr** [comment]
To run these test you can run with:
```
docker run -v ${PWD}/testcafe:/tests --net=host -it testcafe/testcafe chromium:headless /tests/**/*.js
```
Planing to integrate it with SemaphoreCI. We should run the whole platform in `test` mode and then run these tests. @vkotronis Do you think we should have a global test docker compose instead of seperate like we have in backed?

**Comment by vkotronis** [comment]
> To run these test you can run with:
> 
> ```
> docker run -v ${PWD}/testcafe:/tests --net=host -it testcafe/testcafe chromium:headless /tests/**/*.js
> ```
> 
> Planing to integrate it with SemaphoreCI. We should run the whole platform in `test` mode and then run these tests. @vkotronis Do you think we should have a global test docker compose instead of seperate like we have in backed?

@slowr let's have separate for now (as with backend). We can test the frontend and the backend separately that way (if it makes sense).

File: frontend/webapp/render/templates/admin/system.htm [hunk 1]
```diff
@@ -321,7 +321,7 @@ <h1 class="page-header">System</h1>
                 });
                 comment_editor.setSize("100%", 140);
 
-                const ws = new WebSocket('wss://' + window.location.hostname + '/api/graphql', 'graphql-ws');
+                const ws = new WebSocket('wss://' + window.location.host + '/api/graphql', 'graphql-ws');
```
Comment (slowr) [inline frontend/webapp/render/templates/admin/system.htm]: hostname did not carry the port so we changed to host

File: monitor/Dockerfile [hunk 2]
```diff
@@ -18,5 +18,6 @@ RUN mkdir -p /etc/artemis/ && \
     mkdir -p /var/log/artemis/
 
 COPY ./configs/* /etc/artemis/
+COPY ./supervisor.d/* /etc/supervisor/conf.d/
```
Comment (vkotronis) [inline monitor/Dockerfile]: why is this required? Is the default location changed?

File: backend/Dockerfile [hunk 3]
```diff
@@ -23,4 +23,6 @@ RUN sed -i 's/bind 127.0.0.1/bind 0.0.0.0/g' /etc/redis.conf \
     # backwards compatibility
     && ln -s /usr/local/lib/pyenv/versions/3.6.8/bin/python /usr/local/bin
 
+COPY ./supervisor.d/* /etc/supervisor/conf.d/
```
Comment (vkotronis) [inline backend/Dockerfile]: why is this required? Is the default location changed?

Comment (slowr) [reply backend/Dockerfile]: they weren't default configuration so a volume was necessary. I added them on the image so if someone runs the image by itself without mounting it should work

File: monitor/Dockerfile (hunk 2 above)
Comment (slowr) [reply monitor/Dockerfile]: they weren't default configuration so a volume was necessary. I added them on the image so if someone runs the image by itself without mounting it should work

File: backend/Dockerfile (hunk 3 above)
Comment (vkotronis) [reply backend/Dockerfile]: I do not understand, do current users need to change sth in their volume mapping for supervisor confs? Same for monitor. If yes, we need to document it and add new upgrade instructions.

Comment (slowr) [reply backend/Dockerfile]: no. if you run the image without a volume it doesn't have a default supervisor configuration.

File: testcafe/tests/simple-flow.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import { Selector } from 'testcafe';
```
Comment (vkotronis) [inline testcafe/tests/simple-flow.js]: Just add one more advanced test (i.e., with a hijack, anything you want; you can select a testfile example). Just make sure you check all possible fields in hijacks and hijack page. After that, we can merge and we can add tests also for the outdate-fix PR extra. Thanks!

File: docker-compose.testcafe.yaml [hunk 5]
```diff
@@ -0,0 +1,228 @@
+version: '3'
+services:
+    backend:
+        image: artemis_backend
+        container_name: backend
```
Comment (vkotronis) [inline docker-compose.testcafe.yaml]: you also need a 
```
build: ./backend/
```

File: docker-compose.testcafe.yaml [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+            HIJACK_LOG_FILTER: ${HIJACK_LOG_FILTER}
+            MON_TIMEOUT_LAST_BGP_UPDATE: ${MON_TIMEOUT_LAST_BGP_UPDATE}
+            # for backwards compatibility
+            SUPERVISOR_HOST: ${BACKEND_SUPERVISOR_HOST}
+            SUPERVISOR_PORT: ${BACKEND_SUPERVISOR_PORT}
+            HISTORIC: ${HISTORIC}
+            HIJACK_LOG_FIELDS: ${HIJACK_LOG_FIELDS}
+    monitor:
+        image: artemis_monitor
+        container_name: monitor
```
Comment (vkotronis) [inline docker-compose.testcafe.yaml]: you also need a 
```
build: ./monitor/
```

File: docker-compose.testcafe.yaml [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+            BACKEND_SUPERVISOR_HOST: ${BACKEND_SUPERVISOR_HOST}
+            BACKEND_SUPERVISOR_PORT: ${BACKEND_SUPERVISOR_PORT}
+            MON_SUPERVISOR_HOST: ${MON_SUPERVISOR_HOST}
+            MON_SUPERVISOR_PORT: ${MON_SUPERVISOR_PORT}
+            REDIS_HOST: ${REDIS_HOST}
+            REDIS_PORT: ${REDIS_PORT}
+            HISTORIC: ${HISTORIC}
+    frontend:
+        image: artemis_frontend
+        container_name: frontend
```
Comment (vkotronis) [inline docker-compose.testcafe.yaml]: you also need a 
```
build: ./frontend/
```

**Comment by slowr** [comment]
:running: **Benchmark Results** :running:
- BGP Updates (updates/sec): **1285**
- Hijacks Detected (hijacks/sec): **92**
